cli/cmd/upgrade.go

```diff
@@ -381,14 +393,18 @@ func fetchIssuer(k kubernetes.Interface, trustPEM string) (string, string, time.
 	if err != nil {
 		return "", "", time.Time{}, err
 	}
+	if externalIssuer {
```

Comment:
Similar to what the identity service does, we can easily check the for the existence of different keys.  e.g.

```
keyPEM, ok := secret.Data[k8s.IdentityIssuerKeyName]
if !ok {
    keyPEM = secret.Data[k8s.IdentityIssuerKeyNameExternal]
}
```

Reply:
Indeed this is true, and I used to do that right [here](https://github.com/linkerd/linkerd2/pull/3600/commits/470c86a615bf7b01c07b38f67729bbec692ebfaf#diff-7596a8e804160c4bfa0dca95355e8c81R376). But then I read [this here](https://github.com/linkerd/linkerd2/pull/3552#discussion_r336326810) and decided to persist the flag and use it during upgrade. Ironically though, I am not using it in the identity service and I am doing pretty much the opposite of what I am doing here. I am getting lost in all the back and forth. My final opinion is that we have two options: 

1. Never persist it and rely on deducing that from the format of the secret.
2. Persist it and pass a bool everywhere where this decision needs to be made (including as an arg to the identitiy service). 

I am fine with any of these as long as we all agree on one.

Reply:
My preference continues to be that we do not persist it, but I agree with you that it's more important that we be consistent.

Reply:
Let's chat about this at tomorrow's standup. I am starting to wonder if we need the flag at all. It really boils down to are we integrating with the secret structure of an external issuer, or are we just enabling the usage of a pre-created secret of type [`SecretTypeTLS`](https://github.com/kubernetes/api/blob/816a9b7df6780e786246b6e46caadaed8d17e34c/core/v1/types.go#L5490-L5495)? 

Essentially, what cert-manager does is create a secret of that type (just like `kubectl create secret tls my-tls --cert=cert.pem --key=key.pem`, and insert an additional `ca.crt` in there (along with some additional labels and annotations). I think supporting the [`SecretTypeTLS`] secret provides a cleaner integration, because it's a k8s primitive. But it will require us to not depends on it for the trust root.

Reply:
Ummm, yes you are right this is what cert-manager does. I guess we really need to aks ourselves whether we want the benefit of having the ca.crt in the secret. I think its quite useful in fact and if users are using cert-manager specifically then its a good thing to have. Supporting the `SecretTypeTLS` as a cleaner integration path is quite nice especially considering the fact that the format of our secret that we generate by the cli is different than the format of `SecretTypeTLS`, so we might want to change that as well. But isnt all of this a bit out of scope. Not sure, we need to discuss.